Mobile Buy SDK/Mobile Buy SDK/Additions/NSPropertyDescription+BUYAdditions.m

```diff
@@ -0,0 +1,348 @@
+	switch (type) {
+		case NSDateAttributeType:
+			name = BUYDateTransformerName;
+			break;
+			
+		case NSDecimalAttributeType:
+			name = BUYDecimalNumberTransformerName;
+			break;
+			
+		default:
+			break;
+	}
+	return name;
+}
+
+@implementation NSAttributeDescription (BUYAdditions)
+
+- (NSString *)buy_attributeValueClassName
+{
+	return self.userInfo[BUYAttributeValueClassNameKey];
+}
+
+- (Class)buy_attributeValueClass
+{
+	return NSClassFromString([self buy_attributeValueClassName]);
+}
+
+- (NSString *)buy_JSONValueTransformerName
+{
+	NSString *name = [super buy_JSONValueTransformerName];
+	if (name == nil) {
+		name = JSONValueTransformerNameForAttributeType(self.attributeType);
+	}
+	if (name == nil) {
+		name = [[self buy_attributeValueClass] buy_JSONValueTransformerName];
+	}
+	if (name == nil) {
+		name = BUYIdentityTransformerName;
+	}
+	return name;
+}
+
+- (id)buy_valueForJSON:(id)JSON object:(id<BUYObject>)object
+{
+	// An attribute's value is determined by the specified transformer.
+	// An Identity transform returns the same value. `nil` is converted into `NSNull`.
+	id transformedValue = [JSON buy_isValidObject] ? [self.JSONValueTransformer reverseTransformedValue:JSON] : [NSNull null];
+	return transformedValue ?: [NSNull null];
+}
+
+- (id)buy_JSONForValue:(id)object
+{
+	// An attribute's JSON is determined by the specified transformer.
+	// An identify transform returns the same value.
+	return [self.JSONValueTransformer transformedValue:object];
+}
+
+@end
+
+#pragma mark -
+
+@implementation NSRelationshipDescription (BUYAdditions)
+
+#pragma mark - Helpers
+
+- (NSString *)encodesIdSuffix
+{
+	return (self.isToMany ? @"_ids" : @"_id");
+}
+
+// array -> (ordered)set
+- (id)buy_transformArray:(NSArray<id<BUYObject>> *)array
+{
+	return self.isOrdered ? [NSOrderedSet orderedSetWithArray:array] : [NSSet setWithArray:array];
+}
+
+// (ordered)set -> array
+- (NSArray *)buy_arrayForCollection:(id)collection
+{
+	return self.ordered ? [collection array] : [collection allObjects];
+}
+
+// JSON -> model
+- (id)buy_objectForJSON:(id)JSON modelManager:(id<BUYModelManager>)modelManager
+{
+	NSString *entityName = self.destinationEntity.name;
+	// If we are expecting an object `id` the object should already exist.
+	// Otherwise let the object context decide how to resolve the object and update it.
+	if (self.encodesIdInJSON) {
+		return [modelManager buy_objectWithEntityName:entityName identifier:JSON];
+	}
+	else {
+		return [modelManager buy_objectWithEntityName:entityName JSONDictionary:JSON];
+	}
+}
+
+// model -> JSON
+- (id)buy_JSONForObject:(NSObject<BUYObject> *)object
+{
+	// JSON generation for a relationship depends on the rules defined in the model.
+	// The model can explicitly specify using an `id` encoding.
+	// Alternately, if the relationship is compatible, encode the entire object.
+	// We do not encode related objects unless three conditions are satisfied:
+	// 1. the relationship is not many-to-many
```

Comment:
In `-buy_JSONForValue:`. These comments would probably be more appropriate in that method, which invokes this one (for to-one relationships) and others.